### PR TITLE
chore(backend): Add OpenAPI

### PR DESCRIPTION
**Pull request by ermshiperete**
This change adds the @nestjs/swagger module. This allows to use
OpenAPI to document the API. The documentation is available at
http://localhost:3000/api and the OpenAPI JSON spec file at
http://localhost:3000/api-json.

---
This change is [Reviewable](https://reviewable.io/reviews/keymanapp/developer.keyman.com/67)

### CODE REVIEW

File: src/auth/user/user.controller.ts [hunk 1]
```diff
@@ -68,6 +115,22 @@ export class UserController {
...
   // Method used in tests to set the username in the session
   @Get('user/test/:username')
+  @ApiBasicAuth('Tests')
+  @ApiOperation({
+    summary: 'Set username for tests',
+    description: 'Used in tests to set the username in the session',
+  })
+  @ApiHeader({
+    name: 'authorization',
+    description: 'The base64 encoded GitHub username and password',
```
Comment (mcdurdin) [inline src/auth/user/user.controller.ts:125]: This also requires the `Basic` authorization prefix?

File: src/main.ts [hunk 2]
```diff
@@ -37,6 +40,40 @@ async function bootstrap(): Promise<void> {
...
+          scopes: GithubService.scope.split(' '),
+        }
+      }
+    }, 'Default')
+    .addBasicAuth({
+      type: 'http',
+      description: 'Used only in e2e tests. Use GitHub <b>username</b> and <b>personal access token</b>.',
+    }, 'Tests')
+    .addServer('http://localhost:3000', 'Local development server')
+    .addServer('https://keyman-developer-online.azurewebsites.net', 'Main production server')
```
Comment (mcdurdin) [inline src/main.ts:72]: I presume we'll need to tweak this for the future -- I'd like to see this surfaced as a global property somewhere so it is mentioned only once.

File: src/auth/user/user.controller.ts (hunk 1 above)
Comment (ermshiperete) [reply src/auth/user/user.controller.ts:125]: done

File: src/main.ts (hunk 2 above)
Comment (ermshiperete) [reply src/main.ts:72]: I removed the two `.addServer` lines - we don't gain much and they are not needed for Swagger UI to work.

**Pull request merged**